golos.publication/golos.publication.cpp

```diff
@@ -876,4 +896,24 @@ void publication::set_curators_prcnt(structures::mssgid message_id, uint16_t cur
         });
 }
 
+void publication::calcrwrdwt(name account, symbol_code token_code, int64_t mssg_id, base_t post_charge) {
+    auto issuer = token::get_issuer(config::token_name, token_code);
```

Comment:
`issuer` and `token_code` doesn't used in this code -> can be removed.
Add `require_auth(_self)` for disable call this action by user.
For performance optimization you can can calculate, save and send reward weight only if different from 1.